1-development/list_posts_by_rating/src/main/java/com/mixfar/posts/service/PostService.java

```diff
@@ -0,0 +1,32 @@
+package com.mixfar.posts.service;
+
+import com.mixfar.posts.model.Post;
+import com.mixfar.posts.repositories.PostRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service("postService")
+@Transactional
+public class PostService {
```

Comment:
Is there a reason for you not to do the application's logic in this module, by doing it in the ```Controller```?

Reply:
Yes, the controller is responsible for the `Business Logic` so the logic must go there. But we can divide the `Business Logic` into two terms, `Domain Logic` and `Application Logic`.
In my view, the `Domain Logic` should be written into `Service`, and the `Application Logic` should be written on `Controller`